src/fhirtypes/StructureDefinition.ts

```diff
@@ -598,6 +598,29 @@ export class StructureDefinition {
     return matchingSlice;
   }
 
+  findObsoleteChoices(baseElement: ElementDefinition, oldTypes: ElementDefinitionType[]): string[] {
+    // first, find all the elements representing choices for the same choice element
+    const baseId = baseElement.id.slice(0, baseElement.id.lastIndexOf('.'));
```

Comment:
`baseId` is just the id of the parent of `baseElement` right? I think you could use the `parent()` function on the ElementDefinition. It took me a bit to figure out what this was getting, and using `parent()` may make it more clear.

Reply:
Absolutely right, I've basically just copied the first line out of the `parent` method. This is what I get for not reading existing classes closely enough.